labelbox/data/annotation_types/data/raster.py

```diff
@@ -53,37 +87,11 @@ def from_2D_arr(cls, arr: Union[TypedArray[Literal['uint8']],
         return cls(arr=arr, **kwargs)
 
     def bytes_to_np(self, image_bytes: bytes) -> np.ndarray:
-        """
-        Converts image bytes to a numpy array
-        Args:
-            image_bytes (bytes): PNG encoded image
-        Returns:
-            numpy array representing the image
-        """
-        arr = np.array(Image.open(BytesIO(image_bytes)))
-        if len(arr.shape) == 2:
-            arr = np.stack((arr,) * 3, axis=-1)
-        return arr[:, :, :3]
+        return bytes_to_np(image_bytes)
 
     def np_to_bytes(self, arr: np.ndarray) -> bytes:
-        """
-        Converts a numpy array to bytes
-        Args:
-            arr (np.array): numpy array representing the image
-        Returns:
-            png encoded bytes
-        """
-        if len(arr.shape) != 3:
-            raise ValueError(
-                "unsupported image format. Must be 3D ([H,W,C])."
-                f"Use {self.__class__.__name__}.from_2D_arr to construct from 2D"
-            )
-        if arr.dtype != np.uint8:
-            raise TypeError(f"image data type must be uint8. Found {arr.dtype}")
-
-        im_bytes = BytesIO()
-        Image.fromarray(arr).save(im_bytes, format="PNG")
-        return im_bytes.getvalue()
+        self._validate_array(arr)
+        return np_to_bytes(arr)
 
     @property
```

Comment:
This is outside of the changes in the PR so it won't let me make the comment in the right spot but we are still caching the results in memory in addition to disk.
```
arr = self.bytes_to_np(im_bytes)
```